logprep/processor/requester/rule.py

```diff
@@ -0,0 +1,186 @@
+    :caption: Incoming event
+
+    {"message": {"hostname": "BB37293hhj"}}
+
+..  code-block:: json
+    :linenos:
+    :caption: Processed event
+
+    {"message": {"hostname": "BB37293hhj"},
+     "cmdb": {
+         "locaction": {
+             "city": "Montreal",
+             "Building": "L76",
+             "Floor": 3,
+             "Room": 34
+             }
+        }
+    }
+"""
+import inspect
+import json
+import re
+
+import requests
+from attrs import define, field, validators
+
+from logprep.processor.calculator.rule import FIELD_PATTERN
+from logprep.processor.field_manager.rule import FieldManagerRule
+
+parameter_keys = inspect.signature(requests.Request).parameters.keys()
+REQUEST_CONFIG_KEYS = [
+    parameter for parameter in parameter_keys if parameter not in ["hooks", "cookies", "files"]
+] + ["timeout", "proxies", "verify", "cert"]
+
+URL_REGEX_PATTERN = r"(http|https):\/\/.+"
+
+
+HTTP_METHODS = ["GET", "OPTIONS", "HEAD", "POST", "PUT", "PATCH", "DELETE"]
+
+
+class RequesterRule(FieldManagerRule):
+    """Interface for a simple Rule with source_fields and target_field"""
+
+    @define(kw_only=True)
+    class Config(FieldManagerRule.Config):
+        """Config for RequesterRule"""
+
+        source_fields: list = field(
+            factory=list, converter=sorted, validator=validators.instance_of(list)
+        )
+        target_field: str = field(factory=str, validator=validators.instance_of(str))
+        """(Optional) The target field to write the complete response json or body to"""
+        target_field_mapping: dict = field(
+            validator=[
+                validators.instance_of(dict),
+                validators.deep_mapping(
+                    key_validator=validators.instance_of(str),
+                    value_validator=validators.instance_of(str),
+                ),
+            ],
+            factory=dict,
+        )
+        """(Optional) A mapping from dotted_fields to dotted_fields to extract data from response
+        json to target fields. If target_field is given too, this is made additionally"""
+        method: str = field(
+            validator=[
+                validators.instance_of(str),
+                validators.in_(HTTP_METHODS),
+            ]
+        )
+        """The method for the request. must be one of
+        :code:`GET`, :code:`OPTIONS`, :code:`HEAD`,
+        :code:`POST`, :code:`PUT`, :code:`PATCH`, :code:`DELETE`"""
+        url: str = field(
+            validator=[
+                validators.instance_of(str),
+                validators.matches_re(rf"^({URL_REGEX_PATTERN})|({FIELD_PATTERN}.*)"),
+            ]
+        )
+        """The url for the request. You can use dissect pattern language to add field values"""
+        json: dict = field(validator=validators.instance_of(dict), factory=dict)
+        """ (Optional) The json payload. Can be templated by using the pattern
+        :code:`${the.dotted.field}` somewhere in the key or value all elements.
```

Comment:
I would write something like (relevant also for the other docs below): 
```suggestion
        """ (Optional) The json payload. Can be enriched with event data by using the pattern
        :code:`${the.dotted.field}` to retrieve nested field values. 
```

Reply:
You only applied this one to the json data, could you also changes this for the `data` and `params` options please?